airflow/models/taskinstance.py

```diff
@@ -1252,6 +1254,8 @@ class TaskInstance(Base, LoggingMixin):
     queued_dttm = Column(UtcDateTime)
     queued_by_job_id = Column(Integer)
     pid = Column(Integer)
+    # TODO: Should this be nullable?
+    executor = Column(String(1000))
```

Comment:
I believe this should be nullable, but just flagging this to be doublechecked in review.

Reply:
What would the NULL case mean? Compared to, say, an empty string, or the default executor value?

Reply:
Yes. Executor should ratehr be not null and historical value shoudl be set to default executor at migration time.

Reply:
This is actually a nuanced bit that's easy to get confused by. The default must be NULL/empty to maintain the behaviour we have today in the base case. I covered this in the AIP and I think that is the best language I've come up with, so I've attached that below. As well as comment threads from Jens and Jarek who +1 the idea during review:

Explanation from AIP:
![Screenshot from 2024-03-14 11-21-29](https://github.com/apache/airflow/assets/65743084/79de991d-b060-45fd-aeb4-055985de3485)

Comment thread:
![Screenshot from 2024-03-14 11-21-42](https://github.com/apache/airflow/assets/65743084/410cbbf1-26cc-41c3-9ac4-b9458354078b)
![Screenshot from 2024-03-14 11-21-51](https://github.com/apache/airflow/assets/65743084/23984b47-84bd-4030-abe0-e6a0df8d5bb8)
![Screenshot from 2024-03-14 11-22-00](https://github.com/apache/airflow/assets/65743084/89f12dba-98d2-40ad-9ff4-6cc41cd1a311)

Reply:
@potiuk Does the above make sense after I pulled in more context?

Reply:
Ah right. That makes sense (again)